olp-cpp-sdk-core/tests/thread/PriorityQueueExtendedTest.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <queue>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "thread/PriorityQueue.h"
+
+namespace {
+namespace thread = olp::thread;
+
+TEST(PriorityQueueExtendedTest, BaseFunctionality) {
```

Comment:
Please add an additional test, which verifies the order among the same priority requests is kept.
Also, add proper unit tests for all methods to make sure all work properly.